.github/workflows/sonar-scan.yml

```diff
@@ -36,7 +36,8 @@ jobs:
           echo "::set-output name=changed-modules::{ \"include\": $CHANGES }"
 
   run-ci-tests:
-    if: github.event.pull_request.draft == false
+    # Do only run if the PR is not a draft and the changed modules matrix contains at least one entry
+    if: github.event.pull_request.draft == false && fromJson(needs.detect-changes.outputs.changed-modules).include[0] != null
```

Comment:
:information_source: Unfortunately GitHub expressions syntax doesn't support any form of `length`. This seems to be the best way to check if the array has at least one element.